### PR TITLE
Fix stuck WalletProfileState

### PR DESCRIPTION
**Pull request by osdnk**
Fixes RNBW-4424
Figma link (if any):

## What changed (plus any additional context for devs)
I noticed that in this case the callback closing modal is not called, because we're using a bit different route (creation, where previously we were just caring about deriving). I added the callback invocation in this place as well under gated under additional argument) 

## Screen recordings / screenshots
Before:

https://user-images.githubusercontent.com/25709300/187914905-6539ac1b-258d-43a0-9ef5-ae5e082af335.mov

After:

https://user-images.githubusercontent.com/25709300/187914934-f95bf3c0-bae7-48e6-99d4-155b43780b07.mov

## What to test

Happy scenario:
1. Import a wallet via public key / ens
2. Create a wallet from the wallet selection sheet

Other scenarios:
Check all usages of this modal, namely:
1. Importing via seeds at the beginning 
2. Creating a derived wallet 

Check also if RNBW-4247 is gone

### CODE REVIEW

**Review by terrysahaidak — APPROVED** [review]
Works great for me!

**Review by tchayen — APPROVED** [review]
Yessss it works.

**Review by perunt — APPROVED** [review]
LGFM

File: src/screens/ChangeWalletSheet.js [hunk 1]
```diff
@@ -467,7 +467,16 @@ export default function ChangeWalletSheet() {
...
-                    await createWallet(null, color, name);
+                    await createWallet(
+                      null,
+                      color,
+                      name,
+                      false,
+                      null,
+                      null,
+                      false,
+                      true
```
Comment (estebanmino) [inline src/screens/ChangeWalletSheet.js:478]: lgtm

**Review by estebanmino — APPROVED** [review]
code lgtm, tried importing wallet from ens and seedphrase, and creating wallet 🥇

**Review by brunobar79 — APPROVED** [review]

**Review by BrodyHughes — APPROVED** [review]
Looks good to me. Tested several cases:
- watch a wallet then create one.
- import with seed then create one.
- import with secret phrase then create one.
- create a wallet then import.
- creating several subsequent wallets on all above cases.
- naming / not naming the wallets I create.

**Pull request merged**